packages/eslint-plugin/src/rules/no-extra-parens.ts

```diff
@@ -35,21 +35,26 @@ export default util.createRule<Options, MessageIds>({
       const rule = rules.BinaryExpression as (n: typeof node) => void;
 
       // makes the rule think it should skip the left or right
-      if (util.isTypeAssertion(node.left)) {
+      const isLeftTypeAssertion = util.isTypeAssertion(node.left);
+      const isRightTypeAssertion = util.isTypeAssertion(node.right);
+      if (isLeftTypeAssertion && isRightTypeAssertion) {
+        return; // ignore
+      }
+      if (isLeftTypeAssertion) {
         return rule({
           ...node,
           left: {
             ...node.left,
-            type: AST_NODE_TYPES.BinaryExpression as any,
+            type: AST_NODE_TYPES.SequenceExpression as any,
```

Comment:
`SequenceExpression` has highest priority than it will always enforce parens

https://github.com/eslint/eslint/blob/master/lib/rules/utils/ast-utils.js#L768-L769
```
SequenceExpression = 0
```

Reply:
yeah this whole rule extension is super dodgy.
We really need to figure out how eslint can expose some internals for easier extension.